pool_conn.go

```diff
@@ -0,0 +1,93 @@
+package ldap
+
+import (
+	"crypto/tls"
+	"log"
+)
+
+// PoolConn implements Client to override the Close() method
+type PoolConn struct {
+	Conn     Client
+	c        *channelPool
+	unusable bool
+	closeAt  []uint8
+}
+
+func (p *PoolConn) Start() {
+	p.Conn.Start()
+}
+
+func (p *PoolConn) StartTLS(config *tls.Config) error {
+	// FIXME - check if already TLS and then ignore?
+	return p.Conn.StartTLS(config)
+}
+
+// Close() puts the given connects back to the pool instead of closing it.
+func (p *PoolConn) Close() {
+	if p.unusable {
+		log.Printf("Closing unusable connection %v", p.Conn)
+		if p.Conn != nil {
+			p.Conn.Close()
+		}
+		return
+	}
+	p.c.put(p.Conn)
```

Comment:
Need to be super careful about retained state here, like leaving a bound connection to be reused